default/python/AI/PolicyAI.py

```diff
@@ -196,18 +232,38 @@ def _unlocked_by(name: str) -> Optional[str]:
             return diversity
         return None
 
+    _rating_functions = {
+        propaganda: lambda self: 20 + self._rate_opinion(propaganda),
+        algo_research: lambda self: self._rate_algo_research(),
```

Comment:
Let's simplify this.

- exclude self from the arguments.  Since we are inside the class, we don't need to pass it once again.
- replace simple lambdas with the methods.


```suggestion
        algo_research: self._rate_algo_research,
```

Reply:
I'd have to change _rating_functions into a function for this work, which also means that the dict is created anew for each call. The good thing about making self an argument to the lamdas is, that the dict is static.

Reply:
You could move it to `__init__`, so it would be created once.

Reply:
> You could move it to `__init__`, so it would be created once.

That's a good idea, done...